ev3dev2/button.py

```diff
@@ -337,3 +197,268 @@ def backspace(self):
         Check if 'backspace' button is pressed.
         """
         return 'backspace' in self.buttons_pressed
+
+
+# micropython implementation
+if is_micropython():
+
+    try:
+        # This is a linux-specific module.
+        # It is required by the Button class, but failure to import it may be
+        # safely ignored if one just needs to run API tests on Windows.
+        import fcntl
+    except ImportError:
+        log.warning(library_load_warning_message("fcntl", "Button"))
+
+    if platform not in ("ev3", "fake"):
+        raise Exception("micropython button support has not been implemented for '%s'" % platform)
+
+
+    def _test_bit(buf, index):
+        byte = buf[int(index >> 3)]
+        bit = byte & (1 << (index % 8))
+        return bool(bit)
+
+
+    class ButtonBase(ButtonCommon):
```

Comment:
What's this for?

Reply:
The InfraredSensor class inherits from this.

Reply:
yep I tested all the basics on python3 and Micropython. Will look into the CI issues tonight.